src/backend/distributed/commands/alter_table.c

```diff
@@ -1367,6 +1367,11 @@ GetViewCreationCommandsOfTable(Oid relationId)
 		/* See comments on CreateMaterializedViewDDLCommand for its limitations */
 		if (get_rel_relkind(viewOid) == RELKIND_MATVIEW)
 		{
+			if (!includeMatViews)
+			{
+				continue;
+			}
+
```

Comment:
what happens if we encounter with a plain view (later in this loop) that depends on this mat view